pkg/utils/metadata.go

```diff
@@ -44,6 +53,22 @@ func ProjectID(project string, client metadataClient.Client) (string, error) {
 	return projectID, nil
 }
 
+// ProjectIDOrDefault returns the project ID by performing the following order:
+// 1) if the input project ID is valid, simply use it.
+// 2) if there is a PROJECT_ID environmental variable, use it.
+// 3) use metadataClient to resolve project id.
+func ProjectIDOrDefault(ctx context.Context, projectID string) (string, error) {
+	if projectID != "" {
+		return projectID, nil
+	}
+	// Parse project id from env var, if not found we simply continue
+	var env ProjectIDEnvConfig
+	if err := envconfig.Process("", &env); err != nil {
+		logging.FromContext(ctx).Fatal("Failed to process env var", zap.Error(err))
```

Comment:
Calling fatal outside of the init path is frowned upon. Perhaps read it from the env once in an `init()` function, which can call fatal, and use that copy going forward. 

I don't think we plan to allow this value to change at run time, right?

Reply:
This is the original logic in controllers, but I agree that here should be a simple log entry as we have backup plan anyway. 

I am not sure whether we should load it once or not, but it is a question to answer as this function is actually called every time it the reconcilers of all sources are run. Any thoughts @grantr ?

Reply:
Changed Fatal to Error.

Reply:
Changed as suggested, the project id from env var will only be initialized during startup.